Scripts/NodeDataCache.cs

```diff
@@ -60,6 +68,31 @@ public static class NodeDataCache {
                     ports.Add(staticPort.fieldName, port);
                 }
             }
+            
+            // Finally, make sure dynamic port settings correspond to the settings of their backing field
+            foreach (NodePort dynamicPort in dynamicPorts) {
+                string[] parts = dynamicPort.fieldName.Split(' ');
+                if (parts.Length != 2) {
+                    Debug.LogError("Dynamic port name " + dynamicPort.fieldName + " is invalid.");
+                    continue;
+                }
+                string backingPortName = parts[0];
+                NodePort backingPort;
+                if (!staticPorts.TryGetValue(backingPortName, out backingPort)) {
+                    Debug.LogError($"Could not find backing port \"{backingPortName}\" for port {dynamicPort.fieldName}");
+                    continue;
+                }
```

Comment:
This static "backing port" you got here is actually an unwanted side effect, as it is not used for anything. Using it here to your advantage is pretty smart.